### ISSUE TITLE
Exception when using full qualified goals

### ISSUE THREAD

**Issue by tompluess** [egineering-llc/gitflow-helper-maven-plugin#59]
Maven commands fail when using full qualified goals on commandline, when **gitflow-helper-maven-plugin** is configured to have extensions enabled (as described on README.md).

e.g.: `mvn org.codehaus.mojo:versions-maven-plugin:2.1:set`

It fails in #descriptorCreator.findPluginForPrefix(prefix, session) in class MasterPromoteExtension, line 66.

Context: 
This happend in using **gitflow-helper-maven-plugin** in conjunction with the **[gitflow-maven-plugin](https://github.com/aleksandr-m/gitflow-maven-plugin)**, where mvn commands are executed on the commandline.

**Comment by bvarner** [comment]
tl;dr Try adding this to your `pom.xml` and let me know if it resolves the issue. I'll revisit if we should fail the build on line 68 based on an exception from line 66.

```
...
<build>
    <plugins>
    ...
        <plugin>
            <groupId>org.codehaus.mojo</groupId>
            <artifactId>versions-maven-plugin</artifactId>
            <version>2.1</version>
        </plugin>
    ...
    </plugins>
</build>
```

You shouldn't have to define any goals in your pom for the versions plugin, but the extension will want to be able to resolve it.

 -- Longer response below --

I'm assuming you were running from the master branch? 

The only place we modify the execution cycle to remove plugins / goals is if the extension detects that it's running on the master branch.

Judging from what I'm seeing in you bug report, and looking through the source for the [gitflow-maven-plugin source](https://github.com/aleksandr-m/gitflow-maven-plugin/blob/master/src/main/java/com/amashchenko/maven/plugin/gitflow/AbstractGitFlowMojo.java), the gitflow-maven-plugin is effectively doing sub-shell exeuctions of maven to execute alternate plugin goals outside the scope of the initial maven execution.

Since it's referencing plugin goals which don't resolve to plugins in your pom, the sub-execution fails, which is going to cause other problems.

Based on what I'm seeing in your log, you _might_ be able to get away with just adding a plugin section for the org.codehaus.mojo:versions-maven-plugin to your pom as a hold-over for a better fix in this extension. Perhaps we shouldn't fail the master build if we can't resolve the plugin for the arbitrary goal...

**Comment by tompluess** [comment]
thanks for your reply!

actually I added a plugin configuration for the `versions-maven-plugin`, but that does not help since the [gitflow-maven-plugin](https://github.com/aleksandr-m/gitflow-maven-plugin) calls full qualified goals.

However, thanks for removing throwing an exception on unresolvable plugin. I'll try with version 1.7.1

**Comment by bvarner** [comment]
Wow. yeah. I'm not sure why I had that tossing exceptions. I think that was a hold-over from some of the very early trial & error with modifying the plugins that will execute during the build.

Please let me know if it works better, and if it doesn't, if you could give me a really verbose example I'll try to run through it and reproduce, so I know what to address.
